File: RxNewDemo/app/src/main/java/ca/six/demo/dagger/simple/SimpleDemo.java
package ca.six.demo.dagger.simple;

import android.app.Activity;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.widget.TextView;

import javax.inject.Inject;

import ca.six.demo.R;


public class SimpleDemo extends Activity {
    @Inject
    Apple apple;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        DaggerSimpleComponent.builder()
                .build()
                .inject(this);

        System.out.println("szw : apple = " + apple.toString());

    }
}

File: Tools/DaggerPlayground/app/src/main/java/six/ca/dagger101/first/SimpleDemo.kt
package six.ca.dagger101.first

import android.app.Activity
import android.os.Bundle
import android.widget.TextView
import kotlinx.android.synthetic.main.activity_main.*

import javax.inject.Inject

import six.ca.dagger101.R


class SimpleDemo : Activity() {
    @Inject
    @JvmField  //这个要注意. 要说明是public才行, 所以这里加上了一个@JvmField
    var apple: Apple? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        DaggerSimpleComponent.builder()
                .build()
                .inject(this)

        tvMain.text = apple!!.toString()

    }
}